### DISCUSSION TITLE
ClickHouse Liveness Probe Failing (While doing busy operations)

### DISCUSSION THREAD

**Discussion by karthik-thiyagarajan** [ClickHouse/ClickHouse#52597] [Q&A]
We are using "clickhouse/clickhouse:23.3" version deployed in a kube cluster. Very often, my Clickhouse pods (3 shards / 3 replicas - total 9 pods) are in crashloopbackoff state as the liveness probe is failing. When i checked the logs, it is actually doing busy processing some records and throwing exceptions while trying to process them. Any way we can adjust these settings or let ClickHouse choose to prioritize my liveness probe over other requests so that i dont get this restarts very often ?

**Comment by den-crane** [comment]
try `ping` as a probe?

```
curl localhost:8123/ping
Ok.
``

**Reply by karthik-thiyagarajan** [reply]
We are already pinging to get the response, but when ClickHouse is really busy doing lots of operations, it could not respond back on time (with in the timeout of 5 seconds) and thus causing Kubernetes to assume that the node is not responding or active and hence it continuously tries for 10 times in an interval for 10 seconds and thus causing restart of the pod after 1.5 minutes. This is bad, as we are trying to restart the pod, just because the pod is not responding as it is busy doing better work.

**Comment by den-crane** [comment]
```
  --priority arg                                                                       
  Priority of the query. 1 - the highest, higher value - lower priority; 0 - do not use priorities.
```

**Comment by karthik-thiyagarajan** [comment]
Can you tell me where do you want me to set this ?  For now, I've temporarily increased the liveness probe interval and timeout and delayed the restart. Obviously, if there are any other better way, I would be happy to use that. Please help.
